index.js

```diff
@@ -20,4 +20,44 @@ if (typeof window !== 'undefined') {
       navLinks.classList.toggle('active');
     });
   });
-}
\ No newline at end of file
+}
+const contactForm = document.querySelector('.contact-form');
+contactForm.addEventListener('submit', (e) => {
+  alert('Message sent successfully');
+  e.preventDefault();
+  const name = document.querySelector('#name').value;
+  const text = document.querySelector('#text').value;
+  const email = document.querySelector('#email').value;
+  const textarea = document.querySelector('#textarea').value;
+  if (name.length === 0) {
+    document.querySelector('.error-name').innerHTML = 'Please enter a name';
+    document.querySelector('.error-name').style.color = 'red';
+  } else if (text.length === 0) {
+    document.querySelector('.error-text').innerHTML = 'Please enter your Last name';
+    document.querySelector('.error-text').style.color = 'red';
+  } else if (email.length === 0) {
+    document.querySelector('.error-email').innerHTML = 'Please enter a valid email';
+    document.querySelector('.error-email').style.color = 'red';
+  } else if (textarea.length === 0) {
+    document.getElementById('#error_textarea').innerHTML = 'Please enter a query';
+    document.querySelector('#error_textarea').style.color = 'red';
+  } else {
+    // contactForm.submit();
+    // add data to localstorage
+    const messageObject = {
+      visitorname: name,
+      visitorlastname: text,
+      visitoremail: email,
+      message: textarea,
+    };
+    if (localStorage.getItem('messages')) {
+      // append
+      const mymessages = JSON.parse(localStorage.getItem('messages')) || [];
+      mymessages.push(messageObject);
+      localStorage.setItem('messages', JSON.stringify(mymessages));
+    } else {
+    // set localstorage key and value
+      localStorage.setItem('messages', JSON.stringify(messageObject));
+    }
+  }
```

Comment:
- [ ] When the user changes the content of any input field, the data is saved to the local storage kindly implement it. When I enter the data in input field it is not saved in local storage.